src/client/connect/http.rs

```diff
@@ -166,7 +168,23 @@ impl<R> HttpConnector<R> {
     /// Default is `None`.
     #[inline]
     pub fn set_local_address(&mut self, addr: Option<IpAddr>) {
-        self.config_mut().local_address = addr;
+        let cfg = self.config_mut();
+
+        match addr {
```

Comment:
This behavior will allow you to set multiple times, and no longer allows resetting by passing `None`. Instead, it should probably unset the other version, (so setting v4 unsets v6, and vice versa), and unset both if passing `None`.